### PR TITLE
Fix naming of some entities

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Yaroslav Igoshev <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [ ] Resolves #? 
- [ ] tests added and passing

### CODE REVIEW

File: docs/architecture.rst [hunk 1]
```diff
@@ -122,8 +122,8 @@ Modin DataFrame API
 
 * import/export functions
    * ``from_pandas``: Convert a pandas dataframe to a Modin dataframe.
-   * ``to_pandas``: Convert a Modin dataframe to a pandas dataframe.
-   * ``to_numpy``: Convert a Modin dataframe to a numpy array.
+   * ``to_pandas``: Convert a Modin dataframe to a Pandas dataframe.
```
Comment (devin-petersohn) [inline docs/architecture.rst:125]: I think the lowercase `pandas` is what is used by the developers: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html#pandas.DataFrame. That was the motivation for that spelling.

File: modin/pandas/indexing.py [hunk 2]
```diff
@@ -210,7 +210,7 @@ def _write_items(self, row_lookup, col_lookup, item):
 
 
 class _LocIndexer(_LocationIndexerBase):
-    """A indexer for ray_df.loc[] functionality"""
+    """A indexer for modin_df.loc[] functionality"""
```
Comment (devin-petersohn) [inline modin/pandas/indexing.py:213]:
```suggestion
    """An indexer for modin_df.loc[] functionality"""
```

File: modin/pandas/indexing.py [hunk 3]
```diff
@@ -331,7 +331,7 @@ def _compute_lookup(self, row_loc, col_loc):
 
 
 class _iLocIndexer(_LocationIndexerBase):
-    """A indexer for ray_df.iloc[] functionality"""
+    """A indexer for modin_df.iloc[] functionality"""
```
Comment (devin-petersohn) [inline modin/pandas/indexing.py:334]:
```suggestion
    """An indexer for modin_df.iloc[] functionality"""
```

File: modin/pandas/test/test_concat.py [hunk 4]
```diff
@@ -74,14 +74,14 @@ def test_df_concat():
     df_equals(pd.concat([df, df2]), pandas.concat([df, df2]))
 
 
-def test_ray_concat():
+def test_modin_concat():
```
Comment (devin-petersohn) [inline modin/pandas/test/test_concat.py:77]:
```suggestion
def test_concat():
```

Comment (devin-petersohn) [reply modin/pandas/test/test_concat.py:77]: For all of these we can remove `modin`.

File: docs/architecture.rst (hunk 1 above)
Comment (YarShev) [reply docs/architecture.rst:125]: Ok, agree

File: modin/pandas/test/test_concat.py (hunk 4 above)
Comment (YarShev) [reply modin/pandas/test/test_concat.py]: Ok, fixed

**Comment by YarShev** [comment]
@devin-petersohn , CI is red because of looks like something go wrong with codecov again

**Comment by YarShev** [comment]
@devin-petersohn , CI is red because of looks like something go wrong with codecov again

**Comment by YarShev** [comment]
I force-pushed changes in order to rerun CI, but the problem with codecov left.

**Comment by YarShev** [comment]
@devin-petersohn , yes, I merged with master and CI is green

**Pull request merged**